src/Symfony/Component/Cache/Traits/PhpFilesTrait.php

```diff
@@ -133,7 +133,7 @@ protected function doFetch(array $ids)
      */
     protected function doHave($id)
     {
-        if ($this->appendOnly && $this->values[$id]) {
+        if ($this->appendOnly && \array_key_exists($id, $this->values) && $this->values[$id]) {
```

Comment:
`if ($this->appendOnly && isset($this->values[$id])) {` would do it (`null` cannot be found in `$this->values`)

Reply:
Thanks. I didn't know about `null` so I was protecting it as well. I've made that change.